cli/command/service/opts.go

```diff
@@ -236,6 +236,38 @@ func (r *restartPolicyOptions) ToRestartPolicy() *swarm.RestartPolicy {
 	}
 }
 
+type credentialSpecOpt struct {
+	value  *swarm.CredentialSpec
+	source string
+}
+
+func (c *credentialSpecOpt) Set(value string) error {
+	c.source = value
+	c.value = &swarm.CredentialSpec{}
+	switch {
+	case strings.HasPrefix(value, "file://"):
+		c.value.File = strings.TrimPrefix(value, "file://")
```

Comment:
Backstory is here: https://github.com/docker/swarmkit/pull/2075

I proposed to add the content directly, but the review went the other way.

The reasoning is to support `file` and `registry` now (same as `docker run`), and in the future, support `config` (or `secret`)